aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,643 @@
+            # All dimensions get their normal index except for chunk dims
+            # which get None, which tell the get data func to pull the whole dim
+            retrieve_indices = np_index[: len(non_chunk_shape)] + (
+                (None,) * len(chunk_shape)
+            )
+
+            # Fill the numpy array with the delayed arrays
+            lazy_arrays[np_index] = da.from_delayed(
+                delayed(LifReader._get_image_data)(
+                    fs=self._fs,
+                    path=self._path,
+                    scene=self.current_scene_index,
+                    retrieve_dims=blocked_dim_order,
+                    retrieve_indices=retrieve_indices,
+                ),
+                shape=chunk_shape,
+                dtype=sample_plane.dtype,
+            )
+
+        # Convert the numpy array of lazy readers into a dask array
+        image_data = da.block(lazy_arrays.tolist())
+
+        # Because we have set certain dimensions to be chunked and others not
+        # we will need to transpose back to original dimension ordering
+        # Example, if the original dimension ordering was "TZYX" and we
+        # chunked by "T", "Y", and "X"
+        # we created an array with dimensions ordering "ZTYX"
+        transpose_indices = []
+        for i, d in enumerate(selected_scene_dims):
+            new_index = blocked_dim_order.index(d)
+            if new_index != i:
+                transpose_indices.append(new_index)
+            else:
+                transpose_indices.append(i)
+
+        # Only run if the transpose is actually required
+        image_data = da.transpose(image_data, tuple(transpose_indices))
+
+        return image_data
+
+    @staticmethod
+    def _get_coords_and_physical_px_sizes(
+        xml: ET.Element, image_short_info: Dict[str, Any], scene_index: int
+    ) -> Tuple[Dict[str, Any], types.PhysicalPixelSizes]:
+        # Create coord dict
+        coords = {}
+
+        # Get all images
+        img_sets = xml.findall(".//Image")
+
+        # Select the current scene
+        img = img_sets[scene_index]
+
+        # Construct channel list
+        scene_channel_list = []
+        channels = img.findall(".//ChannelDescription")
+        channel_details = img.findall(".//WideFieldChannelInfo")
+        for i, channel in enumerate(channels):
+            if len(channels) <= len(channel_details):
+                channel_detail = channel_details[i]
+                scene_channel_list.append(
+                    (
+                        f"{channel_detail.attrib['LUT']}"
+                        f"--{channel_detail.attrib['ContrastingMethodName']}"
+                        f"--{channel_detail.attrib['FluoCubeName']}"
+                    )
+                )
+            else:
+                scene_channel_list.append(f"{channel.attrib['LUTName']}")
+
+        # Attach channel names to coords
+        coords[DimensionNames.Channel] = scene_channel_list
```

Comment:
i never noticed this, but do the other readers do this too? The channel coords are a list of channel name strings and not a zero-based index?

Reply:
Yep! Coordinate planes can be an iterable of any hashable type. This is also how we are able to keep the reader API so simply. We just call `xarray_dask_data.coords["C"]` to get the channel names for the image. Xarray handles 99% of the data storage and interaction we need.